node/narwhal/src/helpers/committee.rs

```diff
@@ -91,12 +91,12 @@ impl<N: Network> Committee<N> {
 
     /// Returns `true` if the combined stake for the given addresses reaches the quorum threshold.
     /// This method takes in a `HashSet` to guarantee that the given addresses are unique.
-    pub fn is_quorum_threshold_reached(&self, addresses: &HashSet<Address<N>>) -> bool {
+    pub fn is_quorum_threshold_reached(&self, addresses: impl IntoIterator<Item = Address<N>>) -> bool {
```

Comment:
This change should be reverted. See documentation above.

Reply:
hmm, since we only need to ensure that the provided addresses are unique (which is currently the case, as all the sources come from `IndexMap`s and `IndexSet`s), perhaps we could verify the uniqueness with something like [itertools::unique](https://docs.rs/itertools/latest/itertools/trait.Itertools.html#method.unique)?

Reply:
snarkvm_utilities has a uniqueness check. Either way it is O(n) cost

Reply:
indeed, so this would only be a possibly nicer API; I wonder if this is something we could `debug_assert` against in order to avoid runtime penalties (since right now all the use-cases are unique), but I'll revert this for now